src/tsolvers/egraph/EgraphSolver.cc

```diff
@@ -1575,3 +1596,14 @@ void Egraph::printStatistics(std::ostream & os) {
     TSolver::printStatistics(os);
     egraphStats.printStatistics(os);
 }
+
+void Egraph::reanalyze(ERef eref) {
+    // FIXME: This will probably never be false, we need to find a different condition
+    if (backtrack_points.size() > 0) {
+        undo_stack_main.push(Undo(REANALYZE, eref));
+    }
+    if (not enode_store.containsSig(eref)) {
```

Comment:
Is this condition always `false`?  I have the feeling that it might be, but again with no test cases I feel unsure.

Reply:
I believe this can be true. Basically, when at this moment, this term is the root of its equivalence class. In our cases, these are only the newly added interface equalities and it might be there are not equivalent to other equalities at this point. And if you check the case of `REANALYZE` in backtracking, at that point I now remove these from the data structures, so here we need to re-add them.

This might be something that we can optimize here, but I think we can analyze that properly only when we have the full MBTC support.
At this moment, this change does not really have any effect on UF solving (I ran the benchmarks on the cluster).

Reply:
Sorry, I meant the other way!  I noticed that the code in the `if` block is executed in the unit test.

I think it's fine to have it this way for now, but I'd like to not forget to figure whether this is the case, at least by the time we get some examples.  This way maybe I'll remember it better.

Reply:
Right, OK. 
But I think also the other case should be possible: When this equality is *not* the root of its equivalence class and this does not change after the backtracking. Then we don't want to add it back, because it is not the representative.

I will try to add more unit tests later.